File: clj/22.clj
(def names
  (sort (read-string (str "(" (slurp "../input/22") ")"))))

(defn value [name]
  (apply + (map #(- (int %) 64) name)))

(prn (apply +
            (for [n (range (count names))]
              (* (+ 1 n) (value (nth names n))))))

File: hs/22.hs
import Data.Char (ord)
import Data.List (sort)

value = foldl1 (+) . map (\c -> ord c - 64)

main = do
    input <- readFile "../input/22"
    let names = sort (read ("[" ++ input ++ "]") :: [String])
    print $ foldl1 (+) $ zipWith (*) [1..length names] $ map value names
